fontforgeexe/cvpalettes.c

```diff
@@ -3791,16 +3791,20 @@ void _CVPaletteActivate(CharView *cv,int force) {
 		SaveOffsets(old->gw,cvlayers,&cvlayersoff);
 	}
 
-    // Now recreate cvtools
-    GDrawDestroyWindow(cvtools);
+    if (palettes_docked || docking_changed) {
```

Comment:
Not sure this makes sense in the execution context -- does your indentation here reflect your not thinking you are inside the `if ( (old = GDrawGetUserData(cvtools))!=cv || force) {` context? And therefore aren't you already protected from superfluous destruction/creation cycles?

Reply:
Nah it's not superfluous, the conundrum is that you only want to recreate the windows if you're in docked mode, because if you recreate it while it's not docked, it will flicker, which is annoying.

So you need to restrict it to recreating only if you're going from docked to another docked window - but if someone has just toggled from not docked to docked, then `palettes_docked` is true - and if that was your only point of reference, you would never recreate the window, which you must do when toggling the docking.

As for the indentation, that's because I'm just using spaces only, vs. tabs + spaces in the old code. Generally in new code I just use spaces, because the old indentation rules are stupid. (there's been discussions in the past of mass reformatting but there's some arguments about it causing issues with git blame - my current thought is just doing it once and for all fwiw).

Reply:
I don't understand why `GDrawGetUserData(cvtools))!=cv` wouldn't be a good test in those cases too, but the guard shouldn't foul anything up so I guess it doesn't matter much.